feat: Use `BlobFormat` and properly support adding raw blobs

Frando opened the pull request:
## Description

With #1496 we also got a `BlobFormat` in more parts of the code base. This extends this to the CLI, bringing with it a few features:

* Stricter types about adding STDIN vs files
* Support adding from STDIN with `iroh blobs add`
* Support adding single blobs without a wrapping collection
* Support getting single blobs properly

## Notes & open questions

<!-- Any notes, remarks or open questions you have to make about the PR. -->

## Change checklist

- [ ] Self-review.
- [ ] Documentation updates if relevant.
- [ ] Tests if relevant.

rklaehn commented on iroh/src/commands.rs:
```diff
@@ -571,12 +585,20 @@ impl BlobCommands {
+                    // node.
+                    true => TicketOption::Print(None),
+                };
+                let source = BlobSource::from_path_or_stdin(path, in_place, true);
```
maybe make an enum instead?

rklaehn commented on iroh/src/commands/add.rs:
```diff
@@ -7,15 +7,107 @@ use std::{
+pub enum TicketOption {
+    /// Do not print an all-in-one ticket
+    None,
+    /// Print an all-in-oone ticket. Optionally include a request token in the ticket.
```
typo oone -> one

rklaehn reviewed (commented):
This should allow adding bare blobs without a wrapping collection. But how do I actually do it, from the CLI?

I tried adding a single file, and it always produces a wrapping collection. I also don't see a CLI option to disable that.

Tried this:
```
cargo run blob add ../testdir_large_files/test1
```

Frando commented:
Alright:
* Rebased on main
* Combined the `add` options into  a struct for more code reuse:
* Added a `--wrap` option and support it throughout the interfaces From the docs:
```rust
    /// Wrap the added file or directory in a collection.
    ///
    /// When adding a single file, without `wrap` the file is added as a single blob and no
    /// collection is created. When enabling `wrap` it also creates a collection with a
    /// single entry, where the entry's name is the filename and the entry's content is blob.
    ///
    /// When adding a directory, a collection is always created.
    /// Without `wrap`, the collection directly contains the entries from the added direcory.
    /// With `wrap`, the directory will be nested so that all names in the collection are
    /// prefixed with the directory name, thus preserving the name of the directory.
    ///
    /// When adding content from STDIN and setting `wrap` you also need to set `filename` to name
    /// the entry pointing to the content from STDIN.
``` 

From my point this is as intuitive as it gets and ready to merge.

rklaehn commented:
So what does this do if you give the path to a single file when doing iroh start? It seems that it wraps in a collection, which is
inconsistent with the default of add.

Other than that, seems to work just fine.
